data/scripts/creaturescripts/player/offline_training.lua

```diff
@@ -50,15 +50,15 @@ function offlineTraining.onLogin(player)
 	end
 
 	text = string.format("%s.", text)
-	player:sendTextMessage(MESSAGE_EVENT_ADVANCE, text)
+	player:sendTextMessage(MESSAGE_OFFLINE_TRAINING, text)
 
 	local vocation = player:getVocation()
 	local promotion = vocation:getPromotion()
 	local topVocation = not promotion and vocation or promotion
-
 	local updateSkills = false
+
 	if table.contains({ SKILL_CLUB, SKILL_SWORD, SKILL_AXE, SKILL_DISTANCE }, offlineTrainingSkill) then
-		local modifier = topVocation:getBaseAttackSpeed() / 1000
+		local modifier = topVocation:getBaseAttackSpeed() / 1000 / configManager.getFloat(configKeys.RATE_OFFLINE_TRAINING_SPEED)
```

Comment:
We already have a pr similar to these changes. Can you check if this will not impact the formula with another pr?

Reply:
I'm talking about only this line that I selected.